src/server/logs/logs.go

```diff
@@ -147,6 +151,35 @@ func (ls LogService) GetLogs(ctx context.Context, request *logs.GetLogsRequest,
 	return nil
 }
 
+func toLogQL(request *logs.GetLogsRequest) (string, error) {
+	if request == nil {
+		return "", errors.New("nil request")
+	}
+	query := request.Query
+	if query == nil {
+		return "", nil
+	}
+	switch query := query.QueryType.(type) {
+	case *logs.LogQuery_User:
+		switch query := query.User.GetUserType().(type) {
+		case *logs.UserLogQuery_Datum:
+			datum := query.Datum
+			return fmt.Sprintf(`{container=~"user|storage"} | json | datumId=%q or datum=%q`, datum, datum), nil
```

Comment:
I don't think `| json` is going to work.  It doesn't work in the case where Loki is configured with the wrong log driver.  If you want to migrate every customer to the correct driver, that would work too, but I think we have to be very liberal in what we expect.  That's why old pachctl does a hack like `" |= 'datumid'"`, and then further filters it after we do our custom parsing.  (Actually it doesn't, but it should.)

Reply:
After team discussion, we’ll address this when we port the remainder of the log parsing code.